src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -306,14 +306,15 @@ static bool CollectCrashDumpWithCreateDump(Process process, string crashDumpPath
             else
             {
                 // Workaround for https://github.com/dotnet/runtime/issues/93321
-                for (int i = 0; i < 5; i++)
+                const int MaxRetries = 5;
+                for (int i = 0; i < MaxRetries; i++)
                 {
                     try
                     {
                         createdump.Kill(entireProcessTree: true);
                         break;
                     }
-                    catch (Exception e)
+                    catch (Exception e) when (i < MaxRetries - 1)
                     {
                         Console.WriteLine($"Process.Kill(entireProcessTree: true) failed with {e}. Retrying.");
```

Comment:
`{e.Message}` instead?

Reply:
`e.ToString()` has stacktrace that is a potentially useful detail. I have made the output formatting better.